lib/Utils/LLVMShorthands.cpp

```diff
@@ -164,12 +165,38 @@ std::string llvmIRToString(const llvm::Value *V) {
   return IRBuffer;
 }
 
+std::string llvmIRToStableString(const llvm::Value *V) {
+  std::string IRBuffer;
+  llvm::raw_string_ostream RSO(IRBuffer);
+  V->print(RSO, getModuleSlotTrackerFor(V));
+
+  boost::trim_left(IRBuffer);
```

Comment:
I'm not 100% sure about the semantics here but, I guess, if it is correct to move the trimming after the flush, you could also use a `StringRef` here and combine it with the code below (meaning, do all the modifications on the `StringRef` and only edit the std::string once before appending).

Reply:
Actually, the llvm::raw_string_ostream is unbuffered by default, so this should not make any difference; however, I have moved the flush-call